config/vlan.py

```diff
@@ -74,57 +99,74 @@ def delete_db_entry(entry_name, db_connector, db_name):
+    if multiple:
+        vid_list = clicommon.multiple_vlan_parser(ctx, vid)
+    else:
+        if not vid.isdigit():
+            ctx.fail("{} is not integer".format(vid))
+        vid_list.append(int(vid))
+    
     config_db = ValidatedConfigDBConnector(db.cfgdb)
     if ADHOC_VALIDATION:
-        if not clicommon.is_vlanid_in_range(vid):
-            ctx.fail("Invalid VLAN ID {} (1-4094)".format(vid))
-
-        if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
-            ctx.fail("{} does not exist".format(vlan))
-
-        intf_table = db.cfgdb.get_table('VLAN_INTERFACE')
-        for intf_key in intf_table:
-            if ((type(intf_key) is str and intf_key == 'Vlan{}'.format(vid)) or # TODO: MISSING CONSTRAINT IN YANG MODEL
-                (type(intf_key) is tuple and intf_key[0] == 'Vlan{}'.format(vid))):
-                ctx.fail("{} can not be removed. First remove IP addresses assigned to this VLAN".format(vlan))
-
-        keys = [ (k, v) for k, v in db.cfgdb.get_table('VLAN_MEMBER') if k == 'Vlan{}'.format(vid) ]
-
-        if keys: # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("VLAN ID {} can not be removed. First remove all members assigned to this VLAN.".format(vid))
-
-        vxlan_table = db.cfgdb.get_table('VXLAN_TUNNEL_MAP')
-        for vxmap_key, vxmap_data in vxlan_table.items():
-            if vxmap_data['vlan'] == 'Vlan{}'.format(vid):
-                ctx.fail("vlan: {} can not be removed. First remove vxlan mapping '{}' assigned to VLAN".format(vid, '|'.join(vxmap_key)) )
+        for vid in vid_list:
+            log.log_info("'vlan del {}' executing...".format(vid))
+            
+            if not clicommon.is_vlanid_in_range(vid):
+                ctx.fail("Invalid VLAN ID {} (2-4094)".format(vid))
+            
+            #Multiple VLANs needs to be referenced
+            vlan = 'Vlan{}'.format(vid)
+
+            #Multiple VLANs needs to be checked
+            if no_restart_dhcp_relay:
+                if is_dhcpv6_relay_config_exist(db, vlan):
+                    ctx.fail("Can't delete {} because related DHCPv6 Relay config is exist".format(vlan))
+
+            if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
+                log.log_info("{} does not exist".format(vlan))
+                ctx.fail("{} does not exist, Aborting!!!".format(vlan))
+
+            intf_table = db.cfgdb.get_table('VLAN_INTERFACE')
+            for intf_key in intf_table:
+                if ((type(intf_key) is str and intf_key == 'Vlan{}'.format(vid)) or  # TODO: MISSING CONSTRAINT IN YANG MODEL
+                        (type(intf_key) is tuple and intf_key[0] == 'Vlan{}'.format(vid))):
+                    ctx.fail("{} can not be removed. First remove IP addresses assigned to this VLAN".format(vlan))
+
+            keys = [(k, v) for k, v in db.cfgdb.get_table('VLAN_MEMBER') if k == 'Vlan{}'.format(vid)]
+
+            if keys:  # TODO: MISSING CONSTRAINT IN YANG MODEL
```

Comment:
same as above about `TODO` comments

Reply:
Please note that these are exisiting YANG Constraints and our proposed changes/YANG Model has not modified any of these constraints.